internal/QueryProcessor.py

```diff
@@ -10,17 +55,79 @@ class QueryProcessor:
+            language_response = self.query_google_language(query)
+            system_prompt = f"""
+    You will answer my question based on the following information:
+    
+    1. Image label information (vision_response): {vision_response}
+    2. Relevant entity information extracted from the question (language_response): {language_response}
+    
+    Please synthesize the above information and accurately respond to my query. If you are uncertain about any aspects, please indicate the reasons for your uncertainty and make reasonable inferences based on the available information.
+    """
+            response = self.query_gpt(query, system_prompt)
+            return response
+        elif query_label == "explicit":
+            return "A query which is inappropriate "
 
     # TODO  this is separate because it would be nice to be able to switch providers based on relevance
     #       for ex, if one provider is better at a task than another, we could use a switch in query_vision()
     def query_google_vision(self, image):
-        pass
+        # https://cloud.google.com/vision/docs/labels?hl=zh-cn#vision_label_detection-python
+        # Link for label detection (for example, identifying 'Coca-Cola')
+        client = vision.ImageAnnotatorClient()
+        content = image.read()
+        image = vision.Image(content=content)
+        response = client.label_detection(image=image)
+        labels = response.label_annotations
+        vision_response = ', '.join([label.description for label in labels])
+        return vision_response
 
     # TODO  System prompt will come from cognition if necessary
     def query_google_language(self, query: str):
-        pass
+        # https://cloud.google.com/natural-language/docs/samples/language-entities-text?hl=zh_cn#language_entities_text-python
+        # Link for analyzing entities in a string (for example, returning information about 'Coca-Cola'). Not sure which Google Language API to use.
+        client = language_v2.LanguageServiceClient()
+        document_type_in_plain_text = language_v2.Document.Type.PLAIN_TEXT
+        language_code = "en"
+        document = {
+            "content": query,
+            "type_": document_type_in_plain_text,
+            "language_code": language_code,
+        }
+        encoding_type = language_v2.EncodingType.UTF8
+        response = client.analyze_entities(
+            request={"document": document, "encoding_type": encoding_type}
+        )
+        result = ""
+        for entity in response.entities:
+            result += f"{entity.name} ({language_v2.Entity.Type(entity.type_).name}), "
+        language_response = result.rstrip(", ")
+        return language_response
 
     # TODO  We'll use the GPT-4o mini model, system prompt will come from cognition work
-    def query_gpt(self, query: str):
-        pass
\ No newline at end of file
+    def query_gpt(self, query: str, system_prompt: str):
```

Comment:
Also, as mentioned on line 129's comment, `system_prompt` is not necessary as a parameter as it will be made available as an instance member. Please ensure you make this same adjustment across all methods.